scripts/install/install-dependencies.sh

```diff
@@ -253,223 +587,288 @@ function install_latest_cfitsio() {
-
   wget "${URL}"
-  tar zxvf "${FN}"
-  cd "${DIR}"
 
-  echo "Building ${DIR}"
-  make
-  make py
+  # Is the file the same as the one used for the current install?
+  if [[ -f "${md5sum_file}" ]] && md5sum --check --status "${md5sum_file}" && \
+    [[ "$(get_installed_astrometry_version)" != "" ]] ; then
+    echo_bar
+    echo
+    echo -n "Checksum matches already installed astrometry version "
+    echo "$(get_installed_astrometry_version), not replacing."
+    cd "${PANDIR}"
+    rm -rf "${SCRATCH_DIR}"
+    return
+  fi
+  local -r tar_md5sum="$(md5sum "${FN}")"
+
+  echo_bar
+  echo
+  echo "Unpacking ${FN}."
+  tar zxf "${FN}"
+
+  if [[ "${ASTROMETRY_VERSION}" == "latest" ]] ; then
+    # We need to know the version that is in the tar file in order to
+    # enter that directory.
+    ASTROMETRY_VERSION=""
+    for version in $(find astrometry.net-* -maxdepth 0 -type d | sed "s/astrometry.net-//")
+    do
+      if [[ "${ASTROMETRY_VERSION}" == "" ]] ; then
+        ASTROMETRY_VERSION="${version}"
+      else
+        echo_bar
+        echo
+        echo "ERROR: Found two version number directories in the tar file!"
+        find astrometry.net-* -maxdepth 0 -type d
+        exit 1
+      fi
+    done
+
+    if [[ ! -f "${md5sum_file}" ]] ; then
+      # We've not yet been able to check if the installed version should
+      # be replaced.
+      if test_installed_astrometry_version ; then
+        return
+      fi
+    fi
+  fi
+
+  echo_bar
+  echo
+  echo "Building astrometry ${ASTROMETRY_VERSION}, logging to ${make_log}"
+
+  cd "astrometry.net-${ASTROMETRY_VERSION}"
+  make >"${make_log}" 2>&1
+
+  echo_bar
+  echo
+  echo "Building astrometry python support, logging to ${make_py_log}"
+  make py >"${make_py_log}" 2>&1
+
+  echo_bar
+  echo
+  echo "Installing built astrometry into directory ${INSTALL_TO},"
+  echo "and logging to ${make_install_log}"
   if [[ -d "${INSTALL_TO}" ]] ; then
     rm -rf "${INSTALL_TO}"
   fi
   mkdir -p "${INSTALL_TO}"
-  echo "Installing into ${INSTALL_TO}"
-  make install "INSTALL_DIR=${INSTALL_TO}"
+  make install "INSTALL_DIR=${INSTALL_TO}" >"${make_install_log}" 2>&1
 
   add_to_PATH "${INSTALL_TO}/bin"
   cd "${PANDIR}"
   rm -rf "${SCRATCH_DIR}"
+
+  echo "${tar_md5sum}" >"${md5sum_file}"
+  local -r installed_version="$(get_installed_astrometry_version)"
+  if [[ "${ASTROMETRY_VERSION}" != "${installed_version}" ]] ; then
+    echo
+    echo "ERROR: version mismatch after installing astrometry!"
+    echo "Expected: ${ASTROMETRY_VERSION}"
+    echo "  Actual: ${installed_version}"
+    exit 1
+  fi
+}
+
+# Installs astrometry if missing or if the wrong version.
+function maybe_install_astrometry() {
+  if ! test_installed_astrometry_version ; then
+    install_astrometry
+  fi
 }
 
+# Run a POCS script for installing the indices used by astrometry
+# to solve fields.
 function install_astrometry_indices() {
+  echo_bar
+  echo
+  echo "Fetching astrometry and other indices."
```

Comment:
Don't we still somewhere need to tell the astrometry.cfg file about the data dir? Or does it default to `data` inside it's base dir?